### PR TITLE
AVRO-3674: [Rust] Pass the correct enclosing namespace when validating and resolving

### PR DESCRIPTION
**Pull request by martin-g**
AVRO-3674

## What is the purpose of the change

* Fix a bug in schema resolution and validation when processing an inner schema with and without its own namespace

## Verifying this change

* New unit tests are added

## Documentation

- Does this pull request introduce a new feature? yes

New public methods are added. Documented with Rustdoc

### CODE REVIEW

File: lang/rust/avro/src/types.rs [hunk 1]
```diff
@@ -2391,4 +2442,151 @@ Field with name '"b"' is not a member of the map items"#,
...
+            field_a: FieldA {
+                enum_a: EnumType::Symbol2,
+                enum_b: EnumType::Symbol1,
+            },
+        };
+
+        let mut ser = Serializer::default();
+        let test_value: Value = msg.serialize(&mut ser).unwrap();
+        assert!(test_value.validate(&schema), "test_value should validate");
+        // TODO (rikheijdens): I believe this should also resolve?
```
Comment (RikHeijdens) [inline lang/rust/avro/src/types.rs:2513]: Looks like this resolves now so I guess we can get rid of the comment :-)
```suggestion
```

**Review by RikHeijdens — APPROVED** [review]
Nice! Thanks for driving this to the finish line.

File: lang/rust/avro/src/types.rs [hunk 2]
```diff
@@ -2391,4 +2442,150 @@ Field with name '"b"' is not a member of the map items"#,
...
+        let test_value: Value = msg.serialize(&mut ser).unwrap();
+        assert!(test_value.validate(&schema), "test_value should validate");
+        assert!(
+            test_value.resolve(&schema).is_ok(),
+            "test_value should resolve"
+        );
+    }
+
+    #[test]
+    fn test_avro_3674_validate_no_namespace_resolution() {
```
Comment (RikHeijdens) [inline lang/rust/avro/src/types.rs:2520]: @martin-g we could get rid of some code duplication by unifying this test-case and the one above as the only difference between the two should be the `namespace` in the schema.

Comment (martin-g) [reply lang/rust/avro/src/types.rs:2520]: I may do it in a separate PR!
I want to do some cleanup to the `_internal` methods too.

**Comment by martin-g** [comment]
Thank you for the report and the contributions, @RikHeijdens !

**Pull request merged**